### DISCUSSION TITLE
Removal of local cached iModel files

### DISCUSSION THREAD

**Discussion by PeterJBassett** [iTwin/itwinjs-core#3951] [Q&A]
Hello - I am developing/maintaining the FutureOn fieldtwin-itwin bridge connector.  I have noticed that after running a connector synchronisation that large iModel bim files are left in the system cache of the application.

Is there a way to configure a bridge connector to clean up these cache files automatically once a synchronisation has completed?

Alternatively, is there any reason for keeping these cached files? Will it do any harm if my application deletes them after the synchronisation has completed (or failed)?

Note that the FutureOn bridge connector runs as a nodejs service on our backend kubernetes cluster. There is the concern that these files are accumulating and taking up large amounts of storage in the server.

**Comment by calebmshafer** [comment]
@jffmarker @jchick-bentley

**Comment by pmconne** [comment]
Are you doing once-and-done synchronization (i.e., convert input files to iModel, upload iModel, never update the iModel again)?
Or are you doing repeated incremental synchronization of the iModel?

**Reply by kabentley** [reply]
I assume (hope) you're not doing a once-and-done synchronization (if so why?) 

Assuming not, the best thing to do is to *never* delete the briefcases you use to generate changesets. It is beyond silly to delete these large files only to re-download them every run. I suspect that's the way the current connector framework works, but if so that should be fixed. If, on the off chance you have more than one VM running your connector jobs, you'll have to make sure only one runs at a time, and then you should download/apply changesets from the other.

@jffmarker?

**Reply by jffmarker** [reply]
I wouldn't be surprised if Bentley's orchestrators played a role in this design... JS Connectors today only run in the cloud, and the cloud runs each job in a clean VM, so it must re-download each and every time. I would think most cloud orchestrators will have to run this way, or otherwise devise their own cache scheme to keep certain briefcases hot. I know with native Connectors we make an effort to re-use the same briefcase ID so as to not leak briefcases (on a per-connection basis, and a single connection cannot run concurrently with itself)... I don't know off-hand if the JS FWK has a mechanism to re-use IDs... do you know @jchick-bentley?

**Reply by kabentley** [reply]
@jffmarker @swwilson-bsi @jchick-bentley Can we please get together to discuss this? 

If connectors must run in the cloud (*far* better if they run on desktops), then it's critical that they **not** "run each job in a clean VM". It's not about re-using briefcase Ids, it's about re-using briefcases. Obviously the FutureOn connector runs multiple time on the same VM, or he wouldn't be asking about deleting them. I also don't understand the issue about locks. What is being locked and why?

**Reply by jchick-bentley** [reply]
@kabentley I just sent a meeting invitation.

**Reply by jchick-bentley** [reply]
@PeterJBassett We'll schedule a call w/ you to help understand your use case better.
